uaclient/entitlements/tests/test_livepatch.py

```diff
@@ -0,0 +1,220 @@
+"""Tests related to uaclient.entitlement.base module."""
+
+import copy
+from io import StringIO
+import mock
+from nose2.tools.params import params
+
+from uaclient import config
+from uaclient.entitlements.livepatch import (
+    LivepatchEntitlement, process_directives)
+from uaclient import status
+from uaclient.testing.helpers import TestCase
+
+
+LIVEPATCH_MACHINE_TOKEN = {
```

Comment:
I would still suggest wrapping these with `types.MappingProxyType`, so we can be sure that there isn't any accidental modification happening.

Reply:
thanks for this. I wrapped in MappingProxyType. This requires our unit tests to also cast back to dict because we can't json serialize (for config.write_cache) and we can't pickle this type for copy.deepcopy calls in unit tests.